spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -611,14 +615,32 @@ def __init__(self, parent=None):
 
     # --- Helper private methods
     # ------------------------------------------------------------------------
-    def process_server_requests(self):
+    def _process_server_requests(self):
         """Process server requests."""
-        # Check if document needs to be updated:
+        # Check if the document needs to be updated:
         if self._document_server_needs_update:
             self.document_did_change()
             self._document_server_needs_update = False
-        for method, params, requires_response in self._pending_server_requests:
+
+        # Completion types
+        doc_completion = CompletionRequestTypes.DOCUMENT_COMPLETION
+        completion_resolve = CompletionRequestTypes.COMPLETION_RESOLVE
+
+        # Send completion requests first to the server to get faster
+        # completions
+        completion_requests = [
+            req for req in self._pending_server_requests
+            if (req[0] == doc_completion or req[0] == completion_resolve)
+        ]
```

Comment:
Should there be two `_pending_server_requests` lists? One with higher priority? I am not sure which approach would be nicer.

Reply:
> Should there be two _pending_server_requests lists?

Perhaps, but I don't know how to handle that. And it'd probably kill the simplicity of your `schedule_request` decorator.

Since we can sort the higher/lower lists here, I think the gain would be small as well.

Reply:
Agreed